### PR TITLE
refactor(smart_mpc): remove `setup.py` and update `CMakeLists.txt`

### PR DESCRIPTION
**Pull request by asei-proxima**
What's changed:

* remove `setup.py`
* change directory structure
* update `CMakeLists.txt`

Motivation: 

Avoid having to follow special setup procedures

### CODE REVIEW

File: control/autoware_smart_mpc_trajectory_follower/CMakeLists.txt [hunk 1]
```diff
@@ -2,10 +2,29 @@ cmake_minimum_required(VERSION 3.14)
...
 autoware_package()
+ament_python_install_package(${PROJECT_NAME})
+find_package(Python3 REQUIRED COMPONENTS Interpreter Development)
+find_package(pybind11_vendor REQUIRED)
+find_package(pybind11 REQUIRED)
+
+execute_process(COMMAND bash -c "pip3 install numba==0.58.1 --force-reinstall")
+execute_process(COMMAND bash -c "pip3 install GPy")
+
+set(PYTHON_INSTALL_DIR "./scripts")
```
Comment (kosuke55) [inline control/autoware_smart_mpc_trajectory_follower/CMakeLists.txt:15]: is this neccesary?

please refer https://github.com/tier4/driving_log_replayer/blob/develop/driving_log_replayer/CMakeLists.txt#L9 
CMakeLists.txt and directory structure.

Comment (kosuke55) [reply control/autoware_smart_mpc_trajectory_follower/CMakeLists.txt:15]: in driving_log_replayer

Comment (asei-proxima) [reply control/autoware_smart_mpc_trajectory_follower/CMakeLists.txt:15]: Please let me confirm what you are saying.

Are you saying that we should have a `requirements.txt` file and change it to list the dependencies there?

Comment (kosuke55) [reply control/autoware_smart_mpc_trajectory_follower/CMakeLists.txt:15]: Oh, sorry, my link pointing to L9 is incorrect.
I was wondering why the driving log replayer does not have `set(PYTHON_INSTALL_DIR “. /scripts")` is missing and this PR needs it.

**Comment by kosuke55** [comment]
please put `*node.py` under `autoware_smart_mpc_trajectory_follower/scripts,` and the other installs under `autoware_smart_mpc_trajectory_follower/autoware_smart_mpc`.

**Comment by kosuke55** [comment]
solve in https://github.com/autowarefoundation/autoware.universe/pull/8268

**Pull request closed without merging**